test/puppetlabs/kitchensink/core_test.clj

```diff
@@ -430,6 +431,20 @@
       (is (= "c3ab8ff13720e8ad9047dd39466b3c8974e592c2fa383d4a3960714caef0c4f2"
              (utf8-string->sha256 "foobar"))))))
 
+(deftest stream-hashing
+  (testing "Computing a SHA-256 hash for an input stream"
+    (testing "should fail if not passed an input stream"
+      (is (thrown? AssertionError (stream->sha256 "what"))))
+
+    (let [stream-fn #(ByteArrayInputStream. (.getBytes "foobar" "UTF-8"))]
+      (testing "should produce a stable hash"
```

Comment:
This test is a bit superfluous as the following test provides even stronger guarantee.

Reply:
I added this to be consistent with other tests. Primarily preserves that there's nothing weird about calling the function multiple times in a row. One of the funny behaviors with inputstream is that each call consumes the stream, so have to open a new one for each.